examples/kg-chat/csv_kg_chat.py

```diff
@@ -0,0 +1,267 @@
+import typer
+from typing import List
+from dotenv import load_dotenv
+from rich.console import Console
+from rich.prompt import Prompt
+from rich import print
+from pandas import DataFrame, read_csv
+
+from langroid.agent.tool_message import ToolMessage
+from langroid.agent.task import Task
+from langroid.utils.configuration import set_global, Settings
+from langroid.language_models.openai_gpt import OpenAIGPTConfig, OpenAIChatModel
+from langroid.agent.special.neo4j.neo4j_chat_agent import (
+    Neo4jChatAgent,
+    Neo4jChatAgentConfig,
+    Neo4jSettings,
+)
+
+console = Console()
+app = typer.Typer()
+
+
+def _load_csv_dataset(csv_location) -> DataFrame:
+    """
+    Load a CSV dataset from a given file path or URL.
+
+    This function reads a CSV file from the specified path or URL into a pandas
+        DataFrame. It handles exceptions that may occur during file reading.
+
+    Returns:
+    DataFrame: A DataFrame containing the data loaded from the CSV file.
+
+    Raises:
+    Exception: An error occurred while trying to read the CSV file.
+
+    Note:
+    The function assumes the CSV file has a header row.
+    """
+    try:
+        df = read_csv(csv_location)
+        return df
+    except Exception as e:
+        raise Exception(f"Error occurred while reading the CSV file: {e}")
+
+
+def _preprocess_dataframe_for_neo4j(
+    df: DataFrame, default_value: str = None, remove_null_rows: bool = True
+) -> DataFrame:
+    """
+    Preprocess a DataFrame for Neo4j import by fixing mismatched quotes in string
+        columns and handling null or missing values.
+
+    Args:
+        df (DataFrame): The DataFrame to be preprocessed.
+        default_value (str, optional): The default value to replace null values.
+        This is ignored if remove_null_rows is True. Defaults to None.
+        remove_null_rows (bool, optional): If True, rows with any null values will
+            be removed.
+        If False, null values will be filled with default_value. Defaults to False.
+
+    Returns:
+        DataFrame: The preprocessed DataFrame ready for Neo4j import.
+    """
+
+    # Fix mismatched quotes in string columns
+    for column in df.select_dtypes(include=["object"]):
+        df[column] = df[column].apply(
+            lambda x: x + '"' if (isinstance(x, str) and x.count('"') % 2 != 0) else x
+        )
+
+    # Handle null or missing values
+    if remove_null_rows:
+        df = df.dropna()
+    else:
+        if default_value is not None:
+            df = df.fillna(default_value)
+
+    return df
+
+
+class CSVNodeGenerator(ToolMessage):
```

Comment:
Rename -> PandasToKGTool
and set `request` to a concise name: `pandas_to_kg` and update the references everywhere.